### ISSUE TITLE
Aspects do not work when using ':my-library.aar' implicit output target 

### ISSUE THREAD

**Issue by jagobagascon** [bazelbuild/rules_android#19]
I don't know if this a problem with the android rules, with bazel or maybe I'm just not understanding something about how do aspects or implicit output targets work.

I'm trying to use aspects on a rule that depends on an `android_library` [implicit output target](https://docs.bazel.build/versions/master/be/android.html#android_library_implicit_outputs): *name*.aar. This is the only way I know to generate an `aar` file but my aspect is not getting called.

I tried using the [example aspect from the documentation](https://docs.bazel.build/versions/master/skylark/aspects.html#advanced-example) and it is behaving the same as mine: it never gets called. Having the following code

```starlark
android_library(
    name = 'lib_name',
    srcs = glob(['*.java']),
    manifest = 'AndroidManifest.xml',
)

file_count_rule(
    name = 'count',
    deps = [':lib_name']
) # =>  WORKS FINE

file_count_rule(
    name = 'count_aar',
    deps = [':lib_name.aar']
) # => FAILS 
````

If I run `bazel build //:count` I get the number of files. But when I call `bazel build //:count_aar` I get the following error:
```<output file target lib_name.aar> doesn't contain declared provider 'FileCountInfo'```

I don't really understand what *implicit output target*s are, or how are they treated. Is this an expected behaviour?

**Comment by ahumesky** [comment]
hi, just curious what the final outcome for this was for you -- is it no longer something you need?

**Comment by jagobagascon** [comment]
I wasn't sure where to ask this, so I made the same question in the main Bazel repo: https://github.com/bazelbuild/bazel/issues/11545.

In short: An implicit target (like `:lib_name.aar`) is just a file, not a rule, and aspects will only attach to rules. So after that clarification I just decided to close this issue as well.

About my specific problem. I was trying to use aspects to collect the dependencies of an `android_library` to create a `pom.xml` and deploy the Android Artifact to a Maven repo. To make it work **I just changed the rule to contain both: `:lib_name` (only for the aspect to run) and `:lib_name.aar`** to get the file I needed. Not the cleanest solution but it works for now.
